ap/selection/test.py

```diff
@@ -1,306 +1,452 @@
+        steps={"Electron": electron_sel},
+        objects={"Electron": electron_indices},
+    )
+
+
+@selector(uses={req_muon, req_electron})
+def lepton_selection_test(events, stats):
+    # example cuts:
+    # - require exactly one lepton with pt>25, eta<2.4 and tight Id
+
+    muon_indices = req_muon(events)
+    electron_indices = req_electron(events)
+    lepton_sel = ak.num(muon_indices, axis=1) + ak.num(electron_indices, axis=1) == 1
+
+    # build and return selection results
+    return SelectionResult(
+        steps={"Lepton": lepton_sel},
+        objects={"Muon": muon_indices, "Electron": electron_indices},
+    )
+
+
+@selector(
+    uses={
+        jet_selection_test, lepton_selection_test, deepjet_selection_test,
+        "LHEWeight.originalXWGTUP",
+    },
+)
+def test(events, stats, config_inst, **kwargs):
+    # example cuts:
+    # - jet_selection_test
+    # - lepton_selection_test
+    # example stats:
+    # - number of events before and after selection
+    # - sum of mc weights before and after selection
+
+    jet_results = jet_selection_test(events, stats)
+    lepton_results = lepton_selection_test(events, stats)
+    deepjet_results = deepjet_selection_test(events, stats)
+
+    # combined event selection after all steps
+    event_sel = jet_results.steps.Jet & lepton_results.steps.Lepton & deepjet_results.steps.Deepjet
+
+    # build and merge selection results
+    results = SelectionResult(main={"event": event_sel})
+    results += jet_results
+    results += lepton_results
+    results += deepjet_results
+
+    # include categories into results
+    category_results = categories(events, config_inst)
+    results += category_results
+
+    # increment stats
+    events_sel = events[event_sel]
+    stats["n_events"] += len(events)
+    stats["n_events_selected"] += ak.sum(event_sel, axis=0)
+    stats["sum_mc_weight"] += ak.sum(events.LHEWeight.originalXWGTUP)
+    stats["sum_mc_weight_selected"] += ak.sum(events_sel.LHEWeight.originalXWGTUP)
+
+    return results
+
+# deltaR cleaning
+
+
+def cleaning_factory(
+    selector_name: AnyStr,
+    to_clean: AnyStr,
+    *clean_against: List[AnyStr],
+) -> Callable:
+    """
+    factory to generate a function with name *selector_name* that cleans the
+    field *to_clean* in the NanoEventArrays agains the field(s) *clean_agains*.
+    First, the necessary column names to construct four-momenta for the
+    different object fields are constructed, i.e. pt, eta, phi and e for the
+    different objects.
+    Finally, the actual selector function is generated, which uses these
+    columns.
+    """
+
+    # compile the list of variables that are necessary for the four momenta
+    # this list is always the same
+    variables_for_lorentzvec = "pt eta phi e".split()
+
+    # sum up all fields aht are to be considered, i.e. the field *to_clean*
+    # and all fields in *clean_against*
+    all_fields = list(clean_against[:])
+    all_fields.append(to_clean)

```

Comment:
```suggestion
    all_fields = clean_against + [to_clean]
```

A matter of taste of course.